pkg/controllermanager/controller/shoot/shoot_control_delete.go

```diff
@@ -210,6 +210,16 @@ func (c *Controller) runDeleteShootFlow(o *operation.Operation) *gardencorev1alp
 			Fn:           flow.TaskFn(botanist.DeleteClusterAutoscaler).RetryUntilTimeout(defaultInterval, defaultTimeout),
 			Dependencies: flow.NewTaskIDs(initializeShootClients),
 		})
+		destroyNetwork = g.Add(flow.Task{
+			Name:         "Destroying shoot networking",
+			Fn:           flow.TaskFn(botanist.DestroyNetwork).RetryUntilTimeout(defaultInterval, defaultTimeout),
+			Dependencies: flow.NewTaskIDs(initializeShootClients),
```

Comment:
```suggestion
			Dependencies: flow.NewTaskIDs(cleanKubernetesResources),
```

We may only destroy the networking after the cluster has been successfully cleaned.